common/environment/environment.hpp

```diff
@@ -66,7 +66,13 @@ enum EnvMachineOS
     MachineOsSize = 4
 };
 
-
+enum RoxieTargetType
+{
+    RTTUnknown = 0,
+    RTTQueries = 1,
+    RTTQueue = 2,
+    RTTBoth = 3
```

Comment:
RTTBoth = RTTQueries | RTTQueue

There are places where testing a bit would simplify the code - I'll add a comment to one case.